packages/hash/frontend/src/pages/[account-slug]/entities/new.page.tsx

```diff
@@ -1,62 +1,55 @@
+import { validateVersionedUri } from "@blockprotocol/type-system-web";
 import { useRouter } from "next/router";
 import { useState } from "react";
-import { useEffectOnceWhen } from "rooks";
-import { useLoggedInUser } from "../../../components/hooks/useAuthenticatedUser";
+import { useAuthenticatedUser } from "../../../components/hooks/useAuthenticatedUser";
 import { PageErrorState } from "../../../components/page-error-state";
 import { getPlainLayout, NextPageWithLayout } from "../../../shared/layout";
-import { mustBeVersionedUri } from "../types/entity-type/util";
 import { EntityPageLoadingState } from "./[entity-uuid].page/entity-page-loading-state";
 import { NewEntityPage } from "./[entity-uuid].page/new-entity-page";
 import { useCreateNewEntityAndRedirect } from "./[entity-uuid].page/shared/use-create-new-entity-and-redirect";
 
-type PageState = "error" | "loading" | "ready";
-
 const Page: NextPageWithLayout = () => {
   const router = useRouter();
-  const { authenticatedUser } = useLoggedInUser();
   const createNewEntityAndRedirect = useCreateNewEntityAndRedirect();
-  const [pageState, setPageState] = useState<PageState>("loading");
-
-  useEffectOnceWhen(() => {
-    const init = async () => {
-      // show error if url slug it's not users shortname, or shortname of one of users orgs
-      const accountSlug = router.query["account-slug"] as string | undefined;
-      const shortname = accountSlug?.split("@")[1];
-
-      const atUsersNamespace = shortname === authenticatedUser?.shortname;
-      const atOrgsNamespace = !!authenticatedUser?.memberOf.find(
-        (val) => val.shortname === shortname,
-      );
+  const [loading, setLoading] = useState(true);
 
-      if (!atOrgsNamespace && !atUsersNamespace) {
-        return setPageState("error");
-      }
-
-      const entityTypeId = router.query["entity-type-id"] as string | undefined;
+  const { authenticatedUser } = useAuthenticatedUser(
+    undefined,
+    true,
+    async () => {
+      const entityTypeId = String(router.query["entity-type-id"]);
+      const idResult = validateVersionedUri(entityTypeId);
 
-      if (entityTypeId) {
+      if (idResult.type === "Ok") {
         try {
-          await createNewEntityAndRedirect(mustBeVersionedUri(entityTypeId));
+          await createNewEntityAndRedirect(idResult.inner);
         } finally {
-          setPageState("ready");
+          setLoading(false);
         }
       } else {
-        setPageState("ready");
+        setLoading(false);
```

Comment:
What do you want to do if `entity-type-id` isn't valid? Do you want to handle it in some way? Do you want to throw? Do you want to strip it from the URL?

Reply:
Updated here https://github.com/hashintel/hash/pull/1492/commits/aaf972345e19220e82c42c8f4b4c3dae23e31f1d